root-sync/base/crossplane/composition-project.yaml

```diff
@@ -55,34 +55,47 @@ spec:
               kind: Project
               metadata:
                 name: "TO BE PATCHED"
+                namespace: default
                 annotations:
                   cnrm.cloud.google.com/auto-create-network: "false"
               spec:
                 name: "TO BE PATCHED"
-                resourceID: "TO BE PATCHED"
+                # resourceID: "TO BE PATCHED" -- xulid is too long
                 folderRef:
                   external: "TO BE PATCHED"
                 billingAccountRef:
                   external: "TO BE PATCHED"
           references:
+            - dependsOn:
+                apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+                kind: Folder
+                name: "TO BE PATCHED"
+                namespace: default
+            - patchesFrom:
+                apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
+                kind: Folder
+                name: "TO BE PATCHED"
+                namespace: default
+                fieldPath: "status.folderId"
+              toFieldPath: "spec.folderRef.external"
             - patchesFrom:
                 apiVersion: v1
                 kind: ConfigMap
-                namespace: default
+                namespace: backstage
                 name: backstage-config
                 fieldPath: data.GCP_BILLING_ACCOUNT_ID
               toFieldPath: spec.billingAccountRef.external
       patches:
-        - fromFieldPath: "spec.projectId"
+        - fromFieldPath: "spec.projectName"
           toFieldPath: "spec.forProvider.manifest.metadata.name"
         - fromFieldPath: "spec.projectName"
           toFieldPath: "spec.forProvider.manifest.spec.name"
-        - fromFieldPath: "spec.projectId"
-          toFieldPath: "spec.forProvider.manifest.spec.resourceID"
-        - type: FromCompositeFieldPath # use folderId from status
-          fromFieldPath: "status.folderId"
-          toFieldPath: "spec.forProvider.manifest.spec.folderRef.external"
-      
+        # - fromFieldPath: "spec.projectId" -- xulid is too long
+        #   toFieldPath: "spec.forProvider.manifest.spec.resourceID"
+        - fromFieldPath: "spec.folderName"
+          toFieldPath: "spec.references[0].dependsOn.name"
+        - fromFieldPath: "spec.folderName"
+          toFieldPath: "spec.references[1].patchesFrom.name"
```

Comment:
**Non-blocking question**: Can we try to improve on the indexed reference and use a name? This seems a bit rough to maintain.